### ISSUE TITLE
Paragraph top padding issue in a table cell (show blocks guide)

### ISSUE THREAD

**Issue by pszczesniak** [ckeditor/ckeditor5#14435]
## 📝 Provide detailed reproduction steps (if any)

1.  Go to [http://127.0.0.1:8080/ckeditor5/latest/features/show-blocks.html](http://127.0.0.1:8080/ckeditor5/latest/features/show-blocks.html)
2.  Create a table and in a cell add two paragraphs
3.  Turn on Show blocks

### ✔️ Expected result

![](https://github.com/ckeditor/ckeditor5/assets/7074833/17484383-bd25-4679-91f3-5918e15a5bbf)

### ❌ Actual result

![](https://github.com/ckeditor/ckeditor5/assets/7074833/512ca986-9c04-4521-8e95-d8d1a636afd9)

### ❓ Possible solution

```
.live-snippet .ck.ck-content .table p:first-child {
  padding-top: 0;
}
```

:point\_down: 

```
.live-snippet .ck.ck-content:not( .ck-show-blocks ) .table p:first-child {
  padding-top: 0;
}
```

## 📃 Other details

*   Browser: …
*   OS: …
*   First affected CKEditor version: …
*   Installed CKEditor plugins: …

---

If you'd like to see this fixed sooner, add a 👍 reaction to this post.

**Comment by oleq** [comment]
The `.live-snippet .ck.ck-content .table p:first-child` selector has a specificity of [ 0, 5, 1 ]. 

Instead of fixing the styles in docs, I'd increase the specificity of the show blocks styles to ensure this kind of collision does not happen in different integrations with strong selectors for paragraphs. This issue revealed that the feature fails in some real-life scenarios, although I have to say [ 0, 5, 1 ] is quite a lot. 

## Proposal 1

Currently, show blocks uses `.ck.ck-editor__editable.ck-show-blocks p` which is [ 0, 3, 1 ]. We can get it [ 0, 4, 1 ] by using `.ck.ck-editor__editable.ck-show-blocks[dir=ltr] p`. Then we can add `ck-content` or `ck-editor__editable_inline` like `.ck.ck-editor__editable.ck-editor__editable_inline.ck-show-blocks[dir=ltr] p` for [ 0, 5, 1 ] and let the CSS order do the trick.

BTW, in ckeditor 4, the show blocks selector is `.cke_show_blocks.cke_contents_ltr p:not([contenteditable=false]):not(.cke_show_blocks_off)`, which is [ 0, 4, 1 ].

## Proposal 2

Let's go with `important`. It's a desperate move but if the former fails, we get this option.

**Comment by pszczesniak** [comment]
Resolved by #14440
